packages/esm-patient-list-app/src/patient-list-detail/patient-list-detail.component.tsx

```diff
@@ -0,0 +1,159 @@
+            startDate: formatDate(member?.patient?.startDate),
+            uuid: member?.patient?.uuid,
+          }))
+        : [],
+    [patientListMembers],
+  );
+
+  const headers = useMemo(
+    () => [
+      {
+        key: 'name',
+        header: t('name', 'Name'),
+        link: {
+          getUrl: (patient) => `${window.spaBase}/patient/${patient?.uuid}`,
+        },
+      },
+      {
+        key: 'identifier',
+        header: t('identifier', 'Identifier'),
+      },
+      {
+        key: 'sex',
+        header: t('sex', 'Sex'),
+      },
+      {
+        key: 'startDate',
+        header: t('startDate', 'Start Date'),
+      },
+    ],
+    [t],
+  );
+
+  const searchResults = useMemo(
+    () =>
+      patients.filter((patient) =>
+        Object.values(patient).some((value) => value?.toLowerCase().includes(searchString?.toLowerCase())),
+      ),
+    [patients, searchString],
+  );
+
+  const handleSearch = useCallback((str) => {
+    setPageCount(1);
+    setSearchString(str);
+  }, []);
+
+  return (
+    <main className="omrs-main-content">
+      <ExtensionSlot extensionSlotName="breadcrumbs-slot" />
+      <div className={styles.cohortHeader}>
+        <div>
+          {patientListDetails && (
+            <>
+              <h1 className={styles.productiveHeading03}>{patientListDetails?.name}</h1>
+              <h4 className={`${styles.bodyShort02} ${styles.marginTop}`}>{patientListDetails?.description}</h4>
+              <div className={` ${styles.text02} ${styles.bodyShort01} ${styles.marginTop}`}>
+                {patientListDetails?.size} {t('patients', 'patients')} &middot;{' '}
+                <span className={styles.label01}>{t('createdOn', 'Created on')}:</span>{' '}
+                {formatDate(patientListDetails?.startDate ?? '')}
+              </div>
+            </>
+          )}
+        </div>
+        <CustomOverflowMenuComponent
+          menuTitle={
+            <>
+              <span className={styles.actionsButtonText}>{t('actions', 'Actions')}</span>{' '}
+              <OverflowMenuVertical16 style={{ marginLeft: '0.5rem' }} />
+            </>
+          }>
+          <OverflowMenuItem itemText={t('editNameDescription', 'Edit Name/ Description')} />
+        </CustomOverflowMenuComponent>
+      </div>
+      <div className={styles.tableContainer}>
+        <PatientListTable
+          patients={searchResults}
+          columns={headers}
+          isLoading={!patientListMembers && !patients}
+          search={{
+            onSearch: handleSearch,
+            placeHolder: 'Search',
+          }}
+          pagination={{
+            usePagination: true,
+            currentPage,
+            onChange: ({ page, pageSize }) => {
+              if (currentPage !== page) {
+                setPageCount(page);
+              }
+              if (currentPageSize !== pageSize) {
+                setCurrentPageSize(pageSize);
+              }
```

Comment:
You can remove the `if`s here. React won't rerender if you are setting the state to the same value as it is right now (which I assume is what the ifs should prevent).